instrumentation-api/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/rpc/RpcClientMetricsTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.rpc;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.attributeEntry;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.RequestListener;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.testing.InMemoryMetricReader;
+import io.opentelemetry.sdk.testing.assertj.metrics.MetricAssertions;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+
+class RpcClientMetricsTest {
+
+  @Test
+  void collectsMetrics() {
+    InMemoryMetricReader metricReader = new InMemoryMetricReader();
+    SdkMeterProvider meterProvider =
+        SdkMeterProvider.builder().registerMetricReader(metricReader).build();
+
+    RequestListener listener = RpcClientMetrics.get().create(meterProvider.get("test"));
+
+    Attributes requestAttributes =
+        Attributes.builder()
+            .put(SemanticAttributes.RPC_SYSTEM, "grpc")
+            .put(SemanticAttributes.RPC_SERVICE, "myservice.EchoService")
+            .put(SemanticAttributes.RPC_METHOD, "exampleMethod")
+            .build();
+
+    Attributes responseAttributes1 =
+        Attributes.builder()
+            .put(SemanticAttributes.NET_PEER_NAME, "example.com")
+            .put(SemanticAttributes.NET_PEER_IP, "127.0.0.1")
+            .put(SemanticAttributes.NET_PEER_PORT, 8080)
+            .put(SemanticAttributes.NET_TRANSPORT, "ip_tcp")
+            .build();
+
+    Attributes responseAttributes2 =
```

Comment:
WDYT about splitting this into 2 test cases? One for host name, one for IP.

Reply:
Yeah,according to @jsuereth suggestion,we should pefer to net.peer.name, we can fallback to net.peer.ip if net.peer.name is  unavailable.